src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const keyValue = item[key];
+
+    if (!((keyValue as string) in result)) {
+      result[keyValue as string] = [];
+    }
+    result[keyValue as string].push(item);
+  }
```

Comment:
type assertions is unsafe so ensure that keyValue is string (e.g String()) and delete all "as string"